pages/u.tsx

```diff
@@ -60,7 +61,9 @@ export const getServerSideProps: GetServerSideProps<ProfileProps> = async ({ req
 	if (!session) {
 		return {
 			redirect: {
-				destination: '/api/auth/signin',
+				destination: getURLWithParams('/api/auth/signin', {
+					callbackUrl: `${process.env.NEXTAUTH_URL}/u`
```

Comment:
if !session then we are redirecting to u.tsx? Should we redirect to login page?

Reply:
We are redirecting to `/api/auth/signin` with a parameter called `callbackUrl` which helps NextAuth understand that he has to send the user back to u.tsx once login is done.